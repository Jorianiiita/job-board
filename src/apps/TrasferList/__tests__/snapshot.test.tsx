import React from 'react';
import { render } from '@testing-library/react';
import { ItemsColumn } from '..';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useId: () => 'r:id',
}));

describe('ItemsColumn Component', () => {
  const setList = jest.fn();

  it('should match the snapshot', () => {
    const list = new Map([
      ['item1', false],
      ['item2', true],
    ]);
    const { container } = render(
      <ItemsColumn list={list} name="Test Column" setList={setList} />,
    );
    expect(container).toMatchSnapshot();
  });
});
