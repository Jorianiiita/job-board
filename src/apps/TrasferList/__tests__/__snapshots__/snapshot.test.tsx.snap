// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsColumn Component should match the snapshot 1`] = `
<div>
  <div
    class="Test Column coloum"
    data-testid="items-column"
  >
    <form
      aria-label="input-form"
    >
      <input
        aria-label="add new Item in the list"
        name="Test Column-input"
        placeholder="Add new Item"
        type="text"
        value=""
      />
    </form>
    <hr />
    <div>
      <input
        id="r:id"
        type="checkbox"
      />
      <label
        for="r:id"
      >
        1 / 2 Selected
      </label>
    </div>
    <hr />
    <div
      class="items"
      data-testid="Test Column-items"
    >
      <ul>
        <li>
          <div>
            <input
              id="r:id"
              type="checkbox"
            />
            <label
              for="r:id"
            >
              item1
            </label>
          </div>
        </li>
        <li>
          <div>
            <input
              checked=""
              id="r:id"
              type="checkbox"
            />
            <label
              for="r:id"
            >
              item2
            </label>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
